### PR TITLE
Fix date format in SES for SDKv2 compatibility

### PR DESCRIPTION
**Pull request by mgagliardo**
Fixes https://github.com/localstack/localstack/issues/3009

Initializing the object SESBackend which is the default object for creating templates and listing them, transforming the input to using isoformat() for ISO 8601 (SDKv2 compatible) and collecting the output with the same ses_backend object.

Also some minor removals on what it seems to be legacy code.

Create:

```
$ aws --debug --endpoint-url=http://localhost:4566 --profile=test ses create-template --template 'TemplateName=hell-world,SubjectPart=subject,TextPart=hello\nworld,HtmlPart=hello<br/>world'
2021-01-12 03:31:59,466 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/1.18.211 Python/3.7.9 Linux/4.19.0-13-amd64 botocore/1.19.51
2021-01-12 03:31:59,466 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['--debug', '--endpoint-url=http://localhost:4566', '--profile=test', 'ses', 'create-template', '--template', 'TemplateName=hell-world,SubjectPart=subject,TextPart=hello\\nworld,HtmlPart=hello<br/>world']
2021-01-12 03:31:59,466 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_scalar_parsers at 0x7fba58e79440>
2021-01-12 03:31:59,467 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function register_uri_param_handler at 0x7fba5937bef0>
2021-01-12 03:31:59,467 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_assume_role_provider_cache at 0x7fba59364950>
2021-01-12 03:31:59,467 - MainThread - botocore.utils - DEBUG - IMDS ENDPOINT: http://169.254.169.254/
2021-01-12 03:31:59,470 - MainThread - botocore.credentials - DEBUG - Skipping environment variable credential check because profile name was explicitly set.
2021-01-12 03:31:59,470 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function attach_history_handler at 0x7fba59059b90>
2021-01-12 03:31:59,470 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /home/gattes/.miniconda3/lib/python3.7/site-packages/botocore/data/ses/2010-12-01/service-2.json
2021-01-12 03:31:59,478 - MainThread - botocore.hooks - DEBUG - Event building-command-table.ses: calling handler functools.partial(<function _remove_commands at 0x7fba58f087a0>, commands_to_remove=['delete-verified-email-address', 'list-verified-email-addresses', 'verify-email-address'])
2021-01-12 03:31:59,478 - MainThread - awscli.customizations.removals - DEBUG - Removing operation: delete-verified-email-address
2021-01-12 03:31:59,478 - MainThread - awscli.customizations.removals - DEBUG - Removing operation: list-verified-email-addresses
2021-01-12 03:31:59,478 - MainThread - awscli.customizations.removals - DEBUG - Removing operation: verify-email-address
2021-01-12 03:31:59,478 - MainThread - botocore.hooks - DEBUG - Event building-command-table.ses: calling handler <function add_waiters at 0x7fba58e88560>
2021-01-12 03:31:59,484 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /home/gattes/.miniconda3/lib/python3.7/site-packages/botocore/data/ses/2010-12-01/waiters-2.json
2021-01-12 03:31:59,486 - MainThread - awscli.clidriver - DEBUG - OrderedDict([('template', <awscli.arguments.CLIArgument object at 0x7fba58cf0c90>)])
2021-01-12 03:31:59,487 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.ses.create-template: calling handler <function add_streaming_output_arg at 0x7fba58e79b90>
2021-01-12 03:31:59,487 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.ses.create-template: calling handler <function add_cli_input_json at 0x7fba59369ef0>
2021-01-12 03:31:59,487 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.ses.create-template: calling handler <function unify_paging_params at 0x7fba58efd5f0>
2021-01-12 03:31:59,496 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /home/gattes/.miniconda3/lib/python3.7/site-packages/botocore/data/ses/2010-12-01/paginators-1.json
2021-01-12 03:31:59,497 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.ses.create-template: calling handler <function add_generate_skeleton at 0x7fba58f56a70>
2021-01-12 03:31:59,499 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.ses.create-template: calling handler <bound method OverrideRequiredArgsArgument.override_required_args of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x7fba58cf0c50>>
2021-01-12 03:31:59,500 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.ses.create-template: calling handler <bound method GenerateCliSkeletonArgument.override_required_args of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x7fba58cf1490>>
2021-01-12 03:31:59,501 - MainThread - botocore.hooks - DEBUG - Changing event name from load-cli-arg.email.create-template.template to load-cli-arg.ses.create-template.template
2021-01-12 03:31:59,501 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.ses.create-template.template: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fba593a4d10>
2021-01-12 03:31:59,502 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.ses.create-template: calling handler <awscli.argprocess.ParamShorthandParser object at 0x7fba593a4f10>
2021-01-12 03:31:59,502 - MainThread - awscli.argprocess - DEBUG - Parsing param --template as shorthand
2021-01-12 03:31:59,502 - MainThread - awscli.arguments - DEBUG - Unpacked value of 'TemplateName=hell-world,SubjectPart=subject,TextPart=hello\\nworld,HtmlPart=hello<br/>world' for parameter "template": {'TemplateName': 'hell-world', 'SubjectPart': 'subject', 'TextPart': 'hello\\nworld', 'HtmlPart': 'hello<br/>world'}
2021-01-12 03:31:59,502 - MainThread - botocore.hooks - DEBUG - Changing event name from load-cli-arg.email.create-template.cli-input-json to load-cli-arg.ses.create-template.cli-input-json
2021-01-12 03:31:59,502 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.ses.create-template.cli-input-json: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fba593a4d10>
2021-01-12 03:31:59,503 - MainThread - botocore.hooks - DEBUG - Changing event name from load-cli-arg.email.create-template.generate-cli-skeleton to load-cli-arg.ses.create-template.generate-cli-skeleton
2021-01-12 03:31:59,503 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.ses.create-template.generate-cli-skeleton: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7fba593a4d10>
2021-01-12 03:31:59,503 - MainThread - botocore.hooks - DEBUG - Event calling-command.ses.create-template: calling handler <bound method CliInputJSONArgument.add_to_call_parameters of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x7fba58cf0c50>>
2021-01-12 03:31:59,503 - MainThread - botocore.hooks - DEBUG - Event calling-command.ses.create-template: calling handler <bound method GenerateCliSkeletonArgument.generate_json_skeleton of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x7fba58cf1490>>
2021-01-12 03:31:59,503 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: assume-role
2021-01-12 03:31:59,503 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: assume-role-with-web-identity
2021-01-12 03:31:59,503 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: sso
2021-01-12 03:31:59,504 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: shared-credentials-file
2021-01-12 03:31:59,504 - MainThread - botocore.credentials - INFO - Found credentials in shared credentials file: ~/.aws/credentials
2021-01-12 03:31:59,504 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /home/gattes/.miniconda3/lib/python3.7/site-packages/botocore/data/endpoints.json
2021-01-12 03:31:59,516 - MainThread - botocore.hooks - DEBUG - Event choose-service-name: calling handler <function handle_service_name_alias at 0x7fba599bad40>
2021-01-12 03:31:59,518 - MainThread - botocore.hooks - DEBUG - Event creating-client-class.ses: calling handler <function add_generate_presigned_url at 0x7fba59a689e0>
2021-01-12 03:31:59,522 - MainThread - botocore.endpoint - DEBUG - Setting email timeout as (60, 60)
2021-01-12 03:31:59,523 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /home/gattes/.miniconda3/lib/python3.7/site-packages/botocore/data/_retry.json
2021-01-12 03:31:59,523 - MainThread - botocore.client - DEBUG - Registering retry handlers for service: ses
2021-01-12 03:31:59,525 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.ses.CreateTemplate: calling handler <function generate_idempotent_uuid at 0x7fba599ebd40>
2021-01-12 03:31:59,527 - MainThread - botocore.hooks - DEBUG - Event before-call.ses.CreateTemplate: calling handler <function inject_api_version_header_if_needed at 0x7fba599755f0>
2021-01-12 03:31:59,527 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=CreateTemplate) with params: {'url_path': '/', 'query_string': '', 'method': 'POST', 'headers': {'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/1.18.211 Python/3.7.9 Linux/4.19.0-13-amd64 botocore/1.19.51'}, 'body': {'Action': 'CreateTemplate', 'Version': '2010-12-01', 'Template.TemplateName': 'hell-world', 'Template.SubjectPart': 'subject', 'Template.TextPart': 'hello\\nworld', 'Template.HtmlPart': 'hello<br/>world'}, 'url': 'http://localhost:4566/', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x7fba58a65110>, 'has_streaming_input': False, 'auth_type': None}}
2021-01-12 03:31:59,528 - MainThread - botocore.hooks - DEBUG - Event request-created.ses.CreateTemplate: calling handler <bound method RequestSigner.handler of <botocore.signers.RequestSigner object at 0x7fba58a650d0>>
2021-01-12 03:31:59,528 - MainThread - botocore.hooks - DEBUG - Event choose-signer.ses.CreateTemplate: calling handler <function set_operation_specific_signer at 0x7fba599eb290>
2021-01-12 03:31:59,529 - MainThread - botocore.auth - DEBUG - Calculating signature using v4 auth.
2021-01-12 03:31:59,529 - MainThread - botocore.auth - DEBUG - CanonicalRequest:
POST
/

content-type:application/x-www-form-urlencoded; charset=utf-8
host:localhost:4566
x-amz-date:20210112T023159Z

content-type;host;x-amz-date
b16c912c68ff17474ad84796b8cc27181b0b987089ffccf8f73fe3bab09f712b
2021-01-12 03:31:59,529 - MainThread - botocore.auth - DEBUG - StringToSign:
AWS4-HMAC-SHA256
20210112T023159Z
20210112/us-east-1/ses/aws4_request
7d00c8049b188304ece19bf3b071c788645e89247e374f9bc474edfd05d8b7d9
2021-01-12 03:31:59,529 - MainThread - botocore.auth - DEBUG - Signature:
7bb8c5e2c71971d81df114293680423750f76ce7717b7114480a86f80dcbdbed
2021-01-12 03:31:59,530 - MainThread - botocore.endpoint - DEBUG - Sending http request: <AWSPreparedRequest stream_output=False, method=POST, url=http://localhost:4566/, headers={'Content-Type': b'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': b'aws-cli/1.18.211 Python/3.7.9 Linux/4.19.0-13-amd64 botocore/1.19.51', 'X-Amz-Date': b'20210112T023159Z', 'Authorization': b'AWS4-HMAC-SHA256 Credential=1234/20210112/us-east-1/ses/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=7bb8c5e2c71971d81df114293680423750f76ce7717b7114480a86f80dcbdbed', 'Content-Length': '175'}>
2021-01-12 03:31:59,531 - MainThread - urllib3.connectionpool - DEBUG - Starting new HTTP connection (1): localhost:4566
2021-01-12 03:31:59,557 - MainThread - urllib3.connectionpool - DEBUG - http://localhost:4566 "POST / HTTP/1.1" 200 217
2021-01-12 03:31:59,558 - MainThread - botocore.parsers - DEBUG - Response headers: {'content-type': 'text/html; charset=utf-8', 'content-length': '217', 'access-control-allow-origin': '*', 'access-control-allow-methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH', 'access-control-allow-headers': 'authorization,content-type,content-md5,cache-control,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agent,x-amz-target,x-amz-acl,x-amz-version-id,x-localstack-target,x-amz-tagging', 'access-control-expose-headers': 'x-amz-version-id', 'connection': 'close', 'date': 'Tue, 12 Jan 2021 02:31:59 GMT', 'server': 'hypercorn-h11'}
2021-01-12 03:31:59,558 - MainThread - botocore.parsers - DEBUG - Response body:
b'<CreateTemplateResponse xmlns="http://ses.amazonaws.com/doc/2010-12-01/"><CreateTemplateResult/><ResponseMetadata><RequestId>47e0ef1a-9bf2-11e1-9279-0100e8cf12ba</RequestId></ResponseMetadata></CreateTemplateResponse>'
2021-01-12 03:31:59,560 - MainThread - botocore.hooks - DEBUG - Event needs-retry.ses.CreateTemplate: calling handler <botocore.retryhandler.RetryHandler object at 0x7fba58a65550>
2021-01-12 03:31:59,561 - MainThread - botocore.retryhandler - DEBUG - No retry needed.
2021-01-12 03:31:59,562 - MainThread - awscli.formatter - DEBUG - RequestId: 47e0ef1a-9bf2-11e1-9279-0100e8cf12ba
```

Output:
```
$ awslocal --profile=test ses list-templates
TEMPLATESMETADATA	2021-01-12T02:31:59.551596	hell-world
```

### CODE REVIEW

**Comment by mgagliardo** [comment]
Fixed via monkey patch, just iterated over all the templates and format using `isoformat()`

Example:
```
$ awslocal --profile=test ses list-templates
TEMPLATESMETADATA	2021-01-13T14:45:30.770230	hell-world
```

File: tests/integration/test_ses.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    def test_list_templates(self):
+        client = aws_stack.connect_to_service('ses')
+        client.create_template(Template=TEST_TEMPLATE_ATTRIBUTES)
+        templ_list = client.list_templates()['TemplatesMetadata']
+        self.assertEqual(1, len(templ_list))
+        created_template = templ_list[0]
+        self.assertEqual(
+            created_template['Name'], TEST_TEMPLATE_ATTRIBUTES['TemplateName'])
+        self.assertTrue(
```
Comment (whummer) [inline tests/integration/test_ses.py:24]: Looks like the tests are failing on this assertion. I think we could change this to something like:
```
self.assertIn(type(created_template['CreatedTimestamp']), (date, datetime))
```

File: localstack/services/ses/ses_starter.py [hunk 2]
```diff
@@ -66,9 +69,20 @@ def email_responses_send_email(self):
...
     email_responses.send_email = email_responses_send_email
 
+    email_responses_list_templates_orig = email_responses.list_templates
+
+    def list_templates(self):
+        email_templates = ses_backend.list_templates()
+        for template in email_templates:
+            if isinstance(template['Timestamp'], (date, datetime)):
+                # Hack to change the last digits to Java SDKv2 compatible format
+                template['Timestamp'] = template['Timestamp'].isoformat()[:-3] + 'Z'
```
Comment (whummer) [inline localstack/services/ses/ses_starter.py:79]: nitpick: I think we could probably use the `timestamp_millis` util function from `utils/common.py` here:
```
template['Timestamp'] = timestamp_millis(template['Timestamp'])
```

**Review by whummer — COMMENTED** [review]
Looks great, thanks for this PR @mgagliardo 👍 . Only added minor nitpick comments.. Thanks

**Comment by mgagliardo** [comment]
@whummer changes are done. Looks like tests are not passing through though because of some issues with S3..
 
```
======================================================================
FAIL: test_s3_post_object_on_presigned_post (tests.integration.test_s3.S3ListenerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gattes/Documents/repos/github/localstack/tests/integration/test_s3.py", line 544, in test_s3_post_object_on_presigned_post
    self.assertIn(response.status_code, [200, 204])
AssertionError: 400 not found in [200, 204]

======================================================================
FAIL: test_s3_presigned_post_success_action_status_201_response (tests.integration.test_s3.S3ListenerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gattes/Documents/repos/github/localstack/tests/integration/test_s3.py", line 581, in test_s3_presigned_post_success_action_status_201_response
    self.assertEqual(response.status_code, 201)
AssertionError: 400 != 201
```

**Comment by whummer** [comment]
Thanks! 👍

**Pull request merged**